microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,680 @@
+                _40_force_partuuid_file = "/etc/default/grub.d/40-force-partuuid.cfg"
+                if node.shell.exists(_40_force_partuuid_file):
+                    output_40_force_partuuid = node.tools[Cat].read(
+                        _40_force_partuuid_file,
+                        sudo=True
+                    )
+                    modified_files[_40_force_partuuid_file] = output_40_force_partuuid
+
+                    search_pattern = re.compile(r'^GRUB_FORCE_PARTUUID=', re.M)
+                    if find_patterns_in_lines(
+                            output_40_force_partuuid, search_pattern)[0]:
+                        node.tools[Sed].substitute(
+                            'GRUB_FORCE_PARTUUID=.*',
+                            '#GRUB_FORCE_PARTUUID=',
+                            _40_force_partuuid_file,
+                            sudo=True
+                        )
+                        log.debug(
+                            "Commented out GRUB_FORCE_PARTUUID line in "
+                            f"{_40_force_partuuid_file}.")
+                    else:
+                        log.debug("No force part uuid to be replaced, continue test...")
+
+                node.execute(
+                    "update-grub2", sudo=True
+                )
+
+                _entry1 = node.tools[Cat].read_with_filter(
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    'resume=',
+                    sudo=True
+                )
+                assert_that(_entry1).described_as(
+                    f"{_entry1} - Missing config update"
+                    " in 50-cloudimg-settings.cfg file"
+                ).is_true()
+
+                _entry2 = node.tools[Cat].read_with_filter(
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    '^GRUB_HIDDEN_TIMEOUT=30',
+                    sudo=True
+                )
+                assert_that(_entry2).described_as(
+                    f"{_entry2} - Missing config update "
+                    "in 50-cloudimg-settings.cfg file"
+                ).is_true()
+
+                _entry3 = node.tools[Cat].read_with_filter(
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    '^GRUB_TIMEOUT=30',
+                    sudo=True
+                )
+                assert_that(_entry3).described_as(
+                    f"{_entry3} - Missing config update "
+                    "in 50-cloudimg-settings.cfg file"
+                ).is_true()
+
+                log.info(
+                    "Successfully updated 50-cloudimg-settings.cfg file"
+                    " with all three entries.\n"
+                    "Setup Hibernate Kernel Completed.")
+
+        #################################
+        #      SETUP HIBERNATE DONE     #
+        #################################
+
+        # Reboot VM to apply swap setup changes
+        log.info("Rebooting VM")
```

Comment:
It doesn't need a info level log. Check if there is debug log for reboot already. If not, add a debug level log here.